tests/func/test_gc.py

```diff
@@ -36,14 +36,14 @@ def test_gc_api(dvc, good_and_bad_cache):
     dvc.gc(workspace=True)
     odb = dvc.cache.local
     good_cache, bad_cache = good_and_bad_cache
-    assert set(odb.oids_exist([*good_cache, *bad_cache])) == good_cache
+    assert set(odb.all()) == good_cache
```

Comment:
This assertion was passing even before running gc. It wasn't actually retrieving oids from `bad_cache` because it was checking if they exist in the fs, and here they only exist in the odb.